x-pack/plugins/security_solution/public/common/hooks/esql/use_esql_availability.ts

```diff
@@ -9,9 +9,14 @@ import { useMemo } from 'react';
 import { useKibana } from '../../lib/kibana';
 import { useIsExperimentalFeatureEnabled } from '../use_experimental_features';
 
+/**
+ * This hook combines the checks for esql availability within the security solution
+ * If the advanced setting is disabled, ESQL will not be accessible in the UI for any new timeline or new rule creation workflows
+ * The feature flags are still available to provide users an escape hatch in case of any esql related performance issues
+ */
 export const useEsqlAvailability = () => {
   const { uiSettings } = useKibana().services;
-  const isEsqlAdvancedSettingEnabled = uiSettings?.get('discover:enableESQL');
+  const isEsqlAdvancedSettingEnabled = uiSettings?.get('enableESQL');
```

Comment:
maybe we can use const variable according to https://github.com/elastic/kibana/pull/181616#issuecomment-2088720877

Reply:
Yea, agreed. Done, just rushed a little to make BC3, but updated as I think we should be okay 😄